lab-judy/test/gallery-router-test.js

```diff
@@ -0,0 +1,338 @@
+        if (err) return done(err);
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal('new beach');
+        expect(res.body.desc).to.equal('really sunny');
+        done();
+      });
+        });
+
+    //PUT test 400 for passing invalid body
+        it('should return status 400 for passing in invalid body', (done) => {
+          request.put(`${url}/api/gallery/${this.tempGallery._id}`)
+      .set('Content-type', 'application/json')
+      .set({Authorization: `Bearer ${this.tempToken}`})
+      .send('{')
+      .end((err, res) => {
+        expect(res.status).to.equal(400);
+        done();
+      });
+        });
+      });
+
+    //PUT test 401 response w/ 'unauthorized' if bad token given
+      it('should return status 401 and "unauthorized" for bad token or no token given', done => {
+        request.put(`${url}/api/gallery/${this.tempGallery._id}`)
+      .send({name: 'new beach', desc:'really sunny'})
+      .end((err, res) => {
+        expect(res.status).to.equal(401);
+        done();
+      });
+      });
+
+
+    //PUT test 404 for bad id given
+      it('should return status 404 for bad id given', (done) => {
+        request.put(`${url}/api/gallery/badid#`)
+      .set({Authorization: `Bearer ${this.tempToken}`})
+      .end((err, res) => {
+        expect(res.status).to.equal(404);
+        done();
+      });
+      });
+      describe('testing DELETE routes', function(){
+        before(done => {
+          console.log('create user');
+          new User(exampleUser)
+      .generatePasswordHash(exampleUser.password)
+      .then( user => user.save())
+      .then( user => {
+        this.tempUser = user;
+        return user.generateToken();
+      })
+      .then( token => {
+        this.tempToken = token;
+        done();
+      })
+      .catch(done);
+        });
+
+        before( done => {
+          console.log('create gallery');
+          exampleGallery.userID = this.tempUser._id.toString();
+          new Gallery(exampleGallery).save()
+      .then( gallery => {
+        this.tempGallery = gallery;
+        done();
+      })
+      .catch(done);
+        });
+
+        after(() => {
+          delete exampleGallery.userID;
+        });
+
+    //Test DELETE for status 204
+        it('should return status 204 for successful deletion', (done) => {
+          request.delete(`${url}/api/gallery/${this.tempGallery._id}`)
+      .set({Authorization: `Bearer ${this.tempToken}`})
+      .end((err, res) => {
+        if (err) return done (err);
+        expect(res.status).to.equal(204);
+        done();
+      });
+        });
+
+    //Test DELETE for status 404 for bad route
+        it('should return 404 for route not found', (done) => {
+          request.delete(`${url}/api/gallery/badid#`)
+      .set({Authorization: `Bearer ${this.tempToken}`})
+      .end((err, res) => {
+        expect(res.status).to.equal(404);
+        done();
+      });
+        });
+      });
```

Comment:
Ideally, you would also have a test for 401 with Delete, but good job with the bonus tests!